File: .tokenpick-tmp/scan.js
const trending = require('/tmp/trending.json');
const markets = require('/tmp/markets.json');
let dex = {};
try { dex = require('/tmp/dex.json'); } catch(e) {}

const trendingSymbols = new Set(trending.coins.map(c => c.item.symbol.toUpperCase()));
const trendingNames = trending.coins.map(c => `${c.item.symbol.toUpperCase()}(r${c.item.market_cap_rank})`);

let btc24=0, btc7=0, eth24=0, eth7=0;
for (const m of markets) {
  if (m.symbol === 'btc') { btc24 = m.price_change_percentage_24h_in_currency || 0; btc7 = m.price_change_percentage_7d_in_currency || 0; }
  if (m.symbol === 'eth') { eth24 = m.price_change_percentage_24h_in_currency || 0; eth7 = m.price_change_percentage_7d_in_currency || 0; }
}
console.log('TRENDING:', trendingNames.join(', '));
console.log(`BENCH: BTC 24h=${btc24.toFixed(2)}% 7d=${btc7.toFixed(2)}% | ETH 24h=${eth24.toFixed(2)}% 7d=${eth7.toFixed(2)}%`);

const dexSyms = new Set();
if (dex.pairs) {
  for (const p of dex.pairs.slice(0, 60)) {
    const b = (p.baseToken && p.baseToken.symbol) ? p.baseToken.symbol.toUpperCase() : '';
    if (b) dexSyms.add(b);
  }
}

const stables = new Set(['BTC','ETH','USDT','USDC','DAI','BUSD','TUSD','USDE','FDUSD','PYUSD','USDP','USDS','WBTC','WETH','STETH','WSTETH','WEETH','WBETH','SUSDE','USDD','USDX','RLUSD','LBTC','TBTC','EZETH','RSETH','METH','RETH','CBBTC','CBETH','SOLVBTC','BNSOL']);

const scored = [];
for (const m of markets) {
  const sym = m.symbol.toUpperCase();
  if (stables.has(sym)) continue;
  const mcap = m.market_cap || 0;
  if (mcap < 20_000_000) continue;
  const vol = m.total_volume || 0;
  const p24 = m.price_change_percentage_24h_in_currency || 0;
  const p7 = m.price_change_percentage_7d_in_currency || 0;
  let score = 0;
  const bd = [];
  if (p24 > 0) { score += 1; bd.push('24h+'); }
  if (p7 > 0) { score += 1; bd.push('7d+'); }
  if (p24 > 5 && p7 > 5) { score += 2; bd.push('both>5%'); }
  if (trendingSymbols.has(sym)) { score += 2; bd.push('trending'); }
  const vmc = mcap > 0 ? vol/mcap : 0;
  if (vmc >= 0.20) { score += 3; bd.push(`vmc${vmc.toFixed(2)}+3`); }
  else if (vmc >= 0.10) { score += 2; bd.push(`vmc${vmc.toFixed(2)}+2`); }
  if (p7 > btc7 && p7 > eth7) { score += 2; bd.push('RS+2'); }
  if (dexSyms.has(sym)) { score += 1; bd.push('dex+1'); }
  scored.push({ sym, name: m.name, price: m.current_price, mcap, vol, p24, p7, vmc, score, bd });
}

scored.sort((a,b) => b.score - a.score);
console.log('\n===TOP 25 BY SCORE===');
for (const t of scored.slice(0, 25)) {
  console.log(`  ${t.sym.padEnd(10)} ${(''+t.name).padEnd(25)} sc=${t.score} 24h=${t.p24.toFixed(1)}% 7d=${t.p7.toFixed(1)}% vmc=${t.vmc.toFixed(2)} mcap=$${(t.mcap/1e9).toFixed(2)}B vol=$${(t.vol/1e6).toFixed(0)}M [${t.bd.join(',')}]`);
}
console.log('\n===TRENDING COINS (CG) WITH MARKET DATA===');
for (const t of scored) {
  if (trendingSymbols.has(t.sym)) {
    console.log(`  ${t.sym.padEnd(10)} sc=${t.score} 24h=${t.p24.toFixed(1)}% 7d=${t.p7.toFixed(1)}% vmc=${t.vmc.toFixed(2)} mcap=$${(t.mcap/1e6).toFixed(0)}M [${t.bd.join(',')}]`);
  }
}
console.log('\nDEX_TRENDING:', [...dexSyms].slice(0,30).join(','));

File: .tokenpick-tmp/scan.py
#!/usr/bin/env python3
import json

# Load trending
with open('/tmp/trending.json') as f:
    trending = json.load(f)

# Load markets
with open('/tmp/markets.json') as f:
    markets = json.load(f)

# Load dex
with open('/tmp/dex.json') as f:
    dex = json.load(f)

# Build trending symbol set
trending_symbols = set()
trending_names = []
for c in trending.get('coins', []):
    sym = c['item']['symbol'].upper()
    trending_symbols.add(sym)
    trending_names.append(f"{sym} (rank {c['item'].get('market_cap_rank')})")

print("===TRENDING===")
print(", ".join(trending_names))

# BTC and ETH for relative strength
btc_7d = eth_7d = 0
for m in markets:
    if m.get('symbol', '').lower() == 'btc':
        btc_7d = m.get('price_change_percentage_7d_in_currency') or 0
        btc_24h = m.get('price_change_percentage_24h_in_currency') or 0
    if m.get('symbol', '').lower() == 'eth':
        eth_7d = m.get('price_change_percentage_7d_in_currency') or 0
        eth_24h = m.get('price_change_percentage_24h_in_currency') or 0

print(f"\nBENCHMARK: BTC 24h={btc_24h:+.2f}% / 7d={btc_7d:+.2f}%  |  ETH 24h={eth_24h:+.2f}% / 7d={eth_7d:+.2f}%")

# DexScreener trending symbols
dex_trending = set()
for p in dex.get('pairs', [])[:50]:
    base = p.get('baseToken', {}).get('symbol', '').upper()
    if base:
        dex_trending.add(base)

# Score each token
scored = []
for m in markets:
    sym = m.get('symbol', '').upper()
    mcap = m.get('market_cap') or 0
    vol = m.get('total_volume') or 0
    p24 = m.get('price_change_percentage_24h_in_currency') or 0
    p7 = m.get('price_change_percentage_7d_in_currency') or 0
    if mcap < 20_000_000:
        continue
    if sym in ('BTC', 'ETH', 'USDT', 'USDC', 'DAI', 'BUSD', 'TUSD', 'USDE', 'FDUSD', 'PYUSD', 'USDP', 'USDS'):
        continue
    if 'usd' in m.get('id', '').lower() and m.get('id') not in ('uniswap',):
        continue
    score = 0
    breakdown = []
    if p24 > 0:
        score += 1
        breakdown.append('24h+')
    if p7 > 0:
        score += 1
        breakdown.append('7d+')
    if p24 > 5 and p7 > 5:
        score += 2
        breakdown.append('both>5%+2')
    if sym in trending_symbols:
        score += 2
        breakdown.append('trending+2')
    vmc = vol / mcap if mcap > 0 else 0
    if vmc >= 0.20:
        score += 3
        breakdown.append(f'vol/mcap{vmc:.2f}+3')
    elif vmc >= 0.10:
        score += 2
        breakdown.append(f'vol/mcap{vmc:.2f}+2')
    if p7 > btc_7d and p7 > eth_7d:
        score += 2
        breakdown.append('RS+2')
    if sym in dex_trending:
        score += 1
        breakdown.append('dex+1')
    scored.append({
        'sym': sym,
        'name': m.get('name'),
        'price': m.get('current_price'),
        'mcap': mcap,
        'vol': vol,
        'p24': p24,
        'p7': p7,
        'vmc': vmc,
        'score': score,
        'breakdown': breakdown,
    })

scored.sort(key=lambda x: -x['score'])
print("\n===TOP 20 BY SCORE===")
for t in scored[:20]:
    print(f"  {t['sym']:8s} {t['name']:25s} score={t['score']} 24h={t['p24']:+.1f}% 7d={t['p7']:+.1f}% vmc={t['vmc']:.2f} mcap=${t['mcap']/1e9:.2f}B vol=${t['vol']/1e6:.0f}M [{','.join(t['breakdown'])}]")

# Also list the trending coins specifically
print("\n===TRENDING COINS (CG) WITH MARKET DATA===")
for t in scored:
    if t['sym'] in trending_symbols:
        print(f"  {t['sym']:8s} score={t['score']} 24h={t['p24']:+.1f}% 7d={t['p7']:+.1f}% vmc={t['vmc']:.2f} mcap=${t['mcap']/1e6:.0f}M [{','.join(t['breakdown'])}]")
